src/NewTools-SettingsBrowser/Color.extension.st

```diff
@@ -3,8 +3,19 @@ Extension { #name : 'Color' }
 { #category : '*NewTools-SettingsBrowser' }
 Color >> asSettingPresenter: aSettingDeclaration [ 
 
-	^ (StSettingSHStyleButtonColorPresenterItem on: aSettingDeclaration)
-		buttonAction: [ aSettingDeclaration getColor ];
-		whenColorChangedDo: [ self updateColor: aSettingDeclaration color ];
+	| settingPresenter colorValue |
+	
+	colorValue := aSettingDeclaration targetSelector 
+		ifNil: [ aSettingDeclaration realTarget perform: aSettingDeclaration name ]
+		ifNotNil: [ 
+			| settingsTarget |
+			settingsTarget := aSettingDeclaration realTarget perform: aSettingDeclaration targetSelector.
+			settingsTarget perform: aSettingDeclaration getSelector ].
+
+	settingPresenter := (SpPresenter new instantiate: StSettingSHStyleButtonColorPresenterItem on: aSettingDeclaration).
```

Comment:
this is equivallent to just `StSettingSHStyleButtonColorPresenterItem on: aSettingDeclaration`, no need of the bureaucracy... and in any case is not recommended to do it like that. Which is the application that presenter belongs to?
(this still can work if you will after add that presenter to another that knows its app, or set the app, but *as is*, it doesn't) 
The recommended way of doing this is by passing a builder (see `SpPresenterBuilder`)

Reply:
Thanks. I'll address the suggested changes in a new issue